src/displaycontroller.cpp

```diff
@@ -797,6 +814,33 @@ void IRAM_ATTR BitmappedDisplayController::execPrimitive(Primitive const & prim,
     case PrimitiveCmd::CopyToBitmap:
       copyToBitmap(prim.bitmapDrawingInfo);
       break;
+    case PrimitiveCmd::DrawTransformedBitmap: {
+      if (insideISR) {
+        uint32_t cp0_regs[18];
+        // get FPU state
+        uint32_t cp_state = xthal_get_cpenable();
+        
+        if (cp_state) {
+          // Save FPU registers
+          xthal_save_cp0(cp0_regs);
+        } else {
+          // enable FPU
+          xthal_set_cpenable(1);
+        }
+        
+        drawBitmapWithTransform(prim.bitmapTransformedDrawingInfo, updateRect);
+
+        if (cp_state) {
+          // Restore FPU registers
+          xthal_restore_cp0(cp0_regs);
+        } else {
+          // turn it back off
+          xthal_set_cpenable(0);
+        }
```

Comment:
this was the main source of pain for this PR

the nature of vdp-gl is that drawing primitives are executed from inside an ISR (interrupt sub-routine) when the primitives queue is enabled and the user is in a 64 colour single-buffered screen mode.  in all other screen modes the queue is processed using a regular task

when inside an ISR, the FPU may not be enabled, and if it _is_ enabled the FPU state will not be restored when the ISR exits.  this behaviour is not documented anywhere in Espressif's documentation about interrupt allocation.

drawing a transformed bitmap makes use of the FPU, so this was causing a crash.  the reported crash makes no mention of the FPU - instead it looks like a stack corruption issue.  my test code ran in a single-buffered 64 colour screen mode.  it was only after _eventually_ trying out a double-buffered mode and seeing that it would _not_ crash there that I started to find _some_ clues as to what was going on.  not enough clues tho...

fortunately @calc84maniac found a forum post about ISRs crashing when a `float` was being used, and that forum post contained what is essentially this solution

this solution is not _entirely_ optimal.  if there are multiple transformed bitmap plots in the drawing queue then the current solution here will be run for every transformed plot, when in principle it need only be used once.  therefore there is the possibility of a future optimisation to be done here